ocaml/idl/datamodel_pool.ml

```diff
@@ -1023,6 +1035,52 @@ let set_uefi_certificates =
       ]
     ~allowed_roles:_R_POOL_ADMIN ()
 
+let install_rpmgpgkey =
+  call ~name:"install_rpmgpgkey"
+    ~doc:"Install an RPM GPG public key onto the pool."
+    ~params:
+      [
+        (Ref _pool, "self", "The pool")
+      ; (String, "name", "The name of the RPM GPG public key")
+      ; (String, "pubkey", "The RPM GPG public key")
+      ]
+    ~allowed_roles:(_R_POOL_OP ++ _R_CLIENT_CERT)
+    ~lifecycle:[]
+    ~result:(Ref _gpg_key, "The reference of the created Gpg_key object")
+    ()
+
+let uninstall_rpmgpgkey =
+  call ~name:"uninstall_rpmgpgkey"
+    ~doc:"Uninstall an RPM GPG public key from the pool."
+    ~params:
+      [
+        (Ref _pool, "self", "The pool")
+      ; (String, "name", "The name of the RPM GPG public key to be uninstalled")
+      ]
+    ~allowed_roles:(_R_POOL_OP ++ _R_CLIENT_CERT)
+    ~lifecycle:[] ()
+
+let sync_rpmgpgkeys =
+  call ~name:"sync_rpmgpgkeys"
+    ~doc:"Sync all RPM GPG public keys throughout the pool."
+    ~params:
+      [
+        (Ref _pool, "self", "The pool")
+      ; (Set (Ref _host), "hosts", "The hosts to be synced")
+      ]
+    ~allowed_roles:(_R_POOL_OP ++ _R_CLIENT_CERT)
+    ~lifecycle:[] ()
+
+let get_rpm_pubkey_string =
```

Comment:
"string" doesn't have semantic value. Rather use "contents", and the doc state which format are the contents in